### DISCUSSION TITLE
Im trying to figure out are the rfc regulations of ikev2 restrict full encapsulation ?

### DISCUSSION THREAD

**Discussion by killgapwalker** [strongswan/strongswan#523] [General]
Im trying to figure out , are the rfc regulations of ikev2 restrict full encapsulation?
 If im wrong how to configure it ?

**Comment by tobiasbrunner** [comment]
> are the rfc regulations of ikev2 restrict full encapsulation?

What do you mean with "full encapsulation"? And restrict how?

**Reply by killgapwalker** [reply]
I mean that i have encapsulation by ISAKMP and no-filtered traffic through my network adapter and i can even make no filtered traffic enacapsulate, because "i don't know how" ! And i tried iptables like this configurations !
iptables -t nat -A PREROUTING -p tcp -m multiport --sport 1:65535 -j DNAT --to-source 10.6.0.10:4500
iptables -t nat -A POSTROUTING -p tcp -m multiport --dport 1:65535 -j SNAT --to-source 10.6.0.10:4500
I wan't encapsulate everything in my network and filter it to !

**Reply by tobiasbrunner** [reply]
Sorry, I've no idea what you are talking about.

**Reply by killgapwalker** [reply]
Okay, you talking about applications that cant be filtered with protocol at all ? This protocol is only for network layer ? Why than my mobile phone encapsulate everything with Strongswan application ?

**Reply by killgapwalker** [reply]
This is only configuration that need to be switched on.

**Reply by killgapwalker** [reply]
Okay i will try to write algorithm to ESP in ipsec.conf

**Reply by killgapwalker** [reply]
Okay , this is bad answear . 
All initialization going with ESP 
Than i think it is an computer configuration , maybe i need to siwtch forward traffic.
